go/vt/vttablet/tabletserver/vstreamer/engine.go

```diff
@@ -511,3 +513,28 @@ func (vse *Engine) getMySQLEndpoint(ctx context.Context, db dbconfigs.Connector)
 	port, _ := res.Rows[0][1].ToInt64()
 	return fmt.Sprintf("%s:%d", host, port), nil
 }
+
+// mapPKEquivalentCols gets a PK equivalent from mysqld for the table
+// and maps the column names to field indexes in the MinimalTable struct.
+func (vse *Engine) mapPKEquivalentCols(ctx context.Context, table *binlogdatapb.MinimalTable) ([]int, error) {
+	mysqld := mysqlctl.NewMysqld(vse.env.Config().DB)
+	pkeColNames, err := mysqld.GetPrimaryKeyEquivalentColumns(ctx, vse.env.Config().DB.DBName, table.Name)
+	if err != nil {
+		return nil, err
+	}
+	pkeCols := make([]int, len(pkeColNames))
+	matches := 0
+	for n, field := range table.Fields {
+		for i, pkeColName := range pkeColNames {
+			if strings.EqualFold(field.Name, pkeColName) {
+				pkeCols[i] = n
+				matches++
+				break
+			}
+		}
+		if matches == len(pkeColNames) {
+			break
+		}
+	}
+	return pkeCols, nil
+}
```

Comment:
So I'm a bit unsure (as in "I didn't dive deep into this and so don't have a formed opinion") whether this code is needed, or should we reuse the logic in https://github.com/vitessio/vitess/pull/8364/files#diff-5e69c397e15dcfda72bb6fe8dba93b2d11a8d2d7b594ebfbbdcfdc07bfc3791d, which `vreplication` then picks up to populate the pk columns.

I'm mostly concerned that we might end up supporting two different methods of populating non-PK columns in `vreplication`.

Reply:
I had the same concern/thoughts. I have no confidence today in saying that the OnlineDDL `ukColumns` related behavior — getting the PKE or PKE with its own query, passing that info along as table comments, then storing the values in the workflow definition using the `source_unique_key_columns`, `target_unique_key_columns`, and `source_unique_key_target_columns` fields — is no longer needed. I do like having some override mechanism as an outlet for the rare times it may be needed, but otherwise it does feel redundant to me.

I think at least some of that is carried over from Gh-ost?

If you think that much of this could likely be trimmed out to unify OnlineDDL more with other VReplication workflows then I'll create an issue to investigate this and come up with a plan. That will also give me an excuse to finally dive into the OnlineDDL code. 🙂